### PR TITLE
Feature/custom widget

### PR DESCRIPTION
**Pull request by SamuelRioTz**
adding support for custom widgets for home screen
Screen Shot 2021-01-26 at 01 00 02
Screen Shot 2021-01-26 at 01 13 12

### CODE REVIEW

**Comment by mdoelker** [comment]
Code looks good, but could we improve the naming, so it is clear where this custom widget is rendered? In case we add more later we need this to be more specific.

**Comment by SamuelRioTz** [comment]
agree. 
we also may need some extra container, what name should I use
Screen Shot 2021-01-26 at 10 31 40

**Comment by mdoelker** [comment]
hmm maybe `mapOverlayWidget`, `mapExtraFabWidget` (as in Floating Action Button)

**Comment by md-weber** [comment]
This is how the app behaves at the moment.

grafik

File: lib/pages/home.dart [hunk 1]
```diff
@@ -30,6 +29,14 @@ import '../widgets/trufi_drawer.dart';
 
 class HomePage extends StatefulWidget {
   static const String route = '/';
+  final LocaleWidgetBuilder customOverlayWidget;
+  final WidgetBuilder customBetweenFabWidget;
+
+  const HomePage(
+      {Key key,
+      @required this.customOverlayWidget,
```
Comment (md-weber) [inline lib/pages/home.dart:37]: I am not sure if they should be required or optional.

File: lib/pages/home.dart [hunk 2]
```diff
@@ -30,6 +29,14 @@ import '../widgets/trufi_drawer.dart';
...
+      @required this.customBetweenFabWidget})
```
Comment (md-weber) [inline lib/pages/home.dart:38]: I am not sure if they should be required or optional.

File: lib/plan/plan_empty.dart [hunk 3]
```diff
@@ -37,16 +49,39 @@ class PlanEmptyPageState extends State<PlanEmptyPage>
...
         right: 16.0,
         child: _buildLowerActionButtons(context),
       ),
+      Positioned.fill(
+        child: Container(
+          margin: EdgeInsets.only(
+            right: customOverlayWidgetMargin,
+            bottom: 60,
+          ),
+          child: widget.customOverlayWidget != null
```
Comment (md-weber) [inline lib/plan/plan_empty.dart:68]: We even check against null, so it would make sense to not make them required.

File: lib/plan/plan_map.dart [hunk 4]
```diff
@@ -99,16 +109,40 @@ class PlanMapPageState extends State<PlanMapPage>
...
+        ),
+        Positioned.fill(
+          child: Container(
+            margin: EdgeInsets.only(
+              left:
+                  MediaQuery.of(context).size.width - customOverlayWidgetMargin,
+              bottom: 80,
+              top: 65,
+            ),
+            child: widget.customBetweenFabWidget != null
```
Comment (md-weber) [inline lib/plan/plan_map.dart:141]: Same here

**Review by md-weber — COMMENTED** [review]
I think it is good, it would be great if we remove the required fields for the widgets.

File: lib/pages/home.dart (hunk 2 above)
Comment (mdoelker) [reply lib/pages/home.dart:38]: Optional

File: lib/pages/home.dart (hunk 1 above)
Comment (mdoelker) [reply lib/pages/home.dart:37]: Optional

**Review by md-weber — APPROVED** [review]

**Pull request merged**